src/services/measure.service.js

```diff
@@ -42,4 +45,72 @@ const remove = async args => {
   return baseRemove(args, 'Measure');
 };
 
-module.exports = { create, searchById, remove, update };
+/**
+ * takes a measureReport and a set of required data with which to calculate the measure and
+ * creates new documents for the measureReport and requirements in the appropriate collections
+ * @param {*} args the args object passed in by the user
+ * @param {*} req the request object passed in by the user
+ * @returns a transaction-response bundle
+ */
+const submitData = async (args, { req }) => {
+  if (req.body.resourceType !== 'Parameters') {
+    throw new ServerError(null, {
+      statusCode: 400,
+      issue: [
+        {
+          severity: 'error',
+          code: 'BadRequest',
+          details: {
+            text: `Expected 'resourceType: Parameters'. Received 'type: ${req.resourceType}'.`
+          }
+        }
+      ]
+    });
+  }
+  if (!req.body.parameter) {
+    throw new ServerError(null, {
+      statusCode: 400,
+      issue: [
+        {
+          severity: 'error',
+          code: 'BadRequest',
+          details: {
+            text: `Unreadable or empty entity for attribute 'parameter'. Received: ${req.body.parameter}`
+          }
+        }
+      ]
+    });
+  }
+
+  const tb = new TransactionBundle();
+  const resources = req.body.parameter;
+
+  let containsMeasureReport = false;
+
+  resources.forEach(resource => {
+    //TODOMAYBE: add functionality for if resource is itself a bundle
+    if (resource.name === 'measureReport') {
+      containsMeasureReport = true;
+    }
+    tb.addEntryFromResource(resource.resource);
+  });
+  if (!containsMeasureReport) {
+    throw new ServerError(null, {
+      statusCode: 400,
+      issue: [
+        {
+          severity: 'error',
+          code: 'BadRequest',
+          details: {
+            text: `Expected at least one resource with name: 'measureReport' and resourceType: 'MeasureReport.`
+          }
+        }
+      ]
+    });
+  }
+  req.body = tb.toJSON();
+  const output = await uploadTransactionBundle(req, req.res);
```

Comment:
I don't totally understand the res attribute here, so @reviewer take a closer look here

Reply:
With http requests, the `res` attribute refers to the response body. Asymmetrik uses Express under the hood, and the functions that handle the http requests take in both a request (`req`) and a response (`res`). I didn't know that the `res` attribute can belong to `req`, but when doing a `console.log` statement it looks like `req.res` does represent the response object that we need for `uploadTransactionBundle()`.